report.md

```diff
@@ -2,6 +2,24 @@
 
 ## Subsystem 1 (MFA)
 
+Multifactor authentication will apply to all users and will aim for a balance between confidentiality and availability.
+
+###First Factor
+The first step of authentication will involve a username and password. The username will be stored as a hash. The password will be stored as a hash with salt. The salt for each user will be unique and will be generated upon registration. The salt will be stored as plain text.
+
+###Second Factor
+The second factor will be triggered on the completion of the first factor. On passing the first stage a random string will be generated by the server and stored temporarily in the database with the associated user in hash form. An sms will be sent to the user based on the registered phone number with the digits. The user will be redirected to another form to key in the digits. He will have 2 minutes to do so after which the login process will fail and the temporary string will be replaced with nonces. On success the user will receive a session and be able to login and the string will be replaced with nonces.

```

Comment:
Is there a reason to do this instead of having everyone using the Tag as 2nd FA?

Reply:
Yup I get the point for availability, but cos this project I don't think we need to focus on availability? More of like lesser things for us to implement.

How about let's check with prof instead :)